### DISCUSSION TITLE
Singleton containers with testcontainers-node?

### DISCUSSION THREAD

**Discussion by dmurat** [testcontainers/testcontainers-node#692] [Q&A] [answered]
With testcontainers-node, is it possible to use singleton containers (like described in https://java.testcontainers.org/test_framework_integration/manual_lifecycle_control/#singleton-containers) and expect containers to be cleaned up automatically by ryuk?

**Comment by cristianrgreco** [comment, marked as answer]
Hi @dmurat, implementing a singleton is more a language design pattern rather than a Testcontainers concern. 

If for example you are using Jest as your test framework, you could implement a [global setup script](https://jestjs.io/docs/configuration#globalsetup-string) which starts a container and makes it accessible to your tests, and a [global teardown script](https://jestjs.io/docs/configuration#globalteardown-string) which stops the container.

**Reply by dmurat** [reply]
Thank you
